app/src/main/java/com/example/music_app/ui/screens/playlists/PlaylistsScreen.kt

```diff
@@ -1,28 +1,140 @@
+import com.example.music_app.ui.theme.Background2Light
+import com.example.music_app.ui.theme.PLAYLIST_ITEM_SIZE
+import com.example.music_app.ui.theme.SHAPE_SIZE
+import com.example.music_app.ui.theme.White
 
+@Preview
 @Composable
-fun PlaylistsScreen() {
+fun PlaylistsScreen(
+    viewModel: PlaylistsViewModel = viewModel(factory = PlaylistsViewModel.Factory)
+) {
+
+    val playlists by viewModel.playlistsForDisplay.collectAsState()
+    val isLoading by viewModel.isLoading.collectAsState()
+    val lazyListState: LazyListState = rememberLazyListState()
+
+    LaunchedEffect(lazyListState.firstVisibleItemIndex) {
+        viewModel.isScrollOnEnd(lazyListState.firstVisibleItemIndex)
+    }
+
     Column(
-        modifier = Modifier.fillMaxSize(),
-        verticalArrangement = Arrangement.Center,
-        horizontalAlignment = Alignment.CenterHorizontally
+        modifier = Modifier
+            .fillMaxSize()
+            .background(White)
+            .padding(BASE_PADDING),
+        verticalArrangement = Arrangement.spacedBy(BASE_PADDING),
+        horizontalAlignment = CenterHorizontally
     ) {
-        Text(
-            text = "Playlists Screen",
-            fontSize = 24.sp,
-            fontWeight = FontWeight.Bold,
-            color = Color.Black
-        )
+        Surface(
+            color = Background1Light,
+            shape = RoundedCornerShape(SHAPE_SIZE),
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(BLOCK_SIZE)
+        ) {
+        }
+        Surface(
+            color = Background1Light,
+            shape = RoundedCornerShape(SHAPE_SIZE),
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(BLOCK_SIZE)
+        ) {
+        }
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .fillMaxWidth()
+                .weight(1f)
+                .clip(RoundedCornerShape(SHAPE_SIZE))
+                .background(Background1Light)
+        ) {
+            LazyColumn(
+                state = lazyListState,
+                contentPadding = PaddingValues(BASE_PADDING),
+                verticalArrangement = Arrangement.spacedBy(BASE_PADDING),
+                modifier = Modifier
+                    .fillMaxSize()
+            ) {
+                if (playlists != null) {
+                    items(playlists?.size ?: 0) { index ->
+                        PlaylistItem(
+                            playlist = playlists?.get(index),
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .height(PLAYLIST_ITEM_SIZE)
+                                .clip(RoundedCornerShape(SHAPE_SIZE))
+                                .background(Background2Light)
+                                .padding(BASE_PADDING)
+                        )
+                    }
+                }
+            }
```

Comment:
If you avoid nullability for playlists you can use more convenient `items(playlists)`. Also I would recommend specifying the `key` factory for the list items